harper-core/src/token.rs

```diff
@@ -242,6 +242,11 @@ impl TokenKind {
         matches!(self, TokenKind::Punctuation(Punctuation::At))
     }
 
+    pub fn is_case_separator(&self) -> bool {
```

Comment:
This should probably be two separate functions, but it's probably OK for now. If someone really wants it I can figure out how to do OR in pattern matching.

Reply:
It's probably fine to leave this here, but if you do, add a doc-comment that explains what a "case separator" is and why it's needed.

I would much rather create an OR `Pattern` and use that though, since we need one anyway. It's up to you which you decide to do (laziness is totally fine in this case).